pygmt/src/grdfill.py

```diff
@@ -31,25 +83,40 @@ def grdfill(grid, outgrid: str | None = None, **kwargs) -> xr.DataArray | None:
     replace the hole values. If no holes are found the original unchanged grid is
     returned.
 
-    Full option list at :gmt-docs:`grdfill.html`
+    Full option list at :gmt-docs:`grdfill.html`.
 
     {aliases}
 
     Parameters
     ----------
     {grid}
     {outgrid}
-    mode : str
-        Specify the hole-filling algorithm to use.  Choose from **c** for
-        constant fill and append the constant value, **n** for nearest
-        neighbor (and optionally append a search radius in
-        pixels [default radius is :math:`r^2 = \sqrt{{ X^2 + Y^2 }}`,
-        where (*X,Y*) are the node dimensions of the grid]), or
-        **s** for bicubic spline (optionally append a *tension*
-        parameter [Default is no tension]).
+    constantfill
+        Fill the holes with a constant value. Specify the constant value to use.
+    gridfill
+        Fill the holes with values sampled from another (possibly coarser) grid. Specify
+        the grid (a file name or an :class:`xarray.DataArray`) to use for the fill.
+    neighborfill
+        Fill the holes with the nearest neighbor. Specify the search radius in pixels.
+        If set to ``True``, the default search radius will be used
+        (:math:`r^2 = \sqrt{{n^2 + m^2}}`, where (*n,m*) are the node dimensions of the
+        grid).
+    splinefill
+        Fill the holes with a bicubic spline. Specify the tension value to use. If set
+        to ``True``, no tension will be used.
     hole : float
         Set the node value used to identify a point as a member of a hole [Default is
         NaN].
+    mode : str
+        Specify the hole-filling algorithm to use. Choose from **c** for constant fill
+        and append the constant value, **n** for nearest neighbor (and optionally append
+        a search radius in pixels [default radius is :math:`r^2 = \sqrt{{ X^2 + Y^2 }}`,
+        where (*X,Y*) are the node dimensions of the grid]), or **s** for bicubic spline
+        (optionally append a *tension* parameter [Default is no tension]).
+
+        .. deprecated:: 0.15.0
+            The ``mode`` parameter is deprecated since v0.15.0 and will be removed in
+            v0.19.0. Use ``constantfill/gridfill/neighborfill/splinefill`` instead.
```

Comment:
```suggestion
        .. deprecated:: 0.15.0
            The ``mode`` parameter is deprecated since v0.15.0 and will be removed in
            v0.19.0. Use ``constantfill``, ``gridfill``, ``neighborfill`` or ``splinefill`` instead.
```